### PR TITLE
Add pubkey setup for datacenter nodes

### LINKED ISSUE #5440: Automate user/pubkey addition to data center nodes
We need a way to quickly and without typos create users with passwordless sudo and pubkey based auth so that various engineering team members can SSH to the data center nodes.  

Perhaps a source controlled script that one of us can run that adds this info to each server in kind.  

Also should consider where to store the pubkeys of the team members such that a malicious actor cannot edit a pubkey and then gain access to the systems.

Would also be nice to have an analogous feature to remove access from all systems in the event someone no longer needs to use them or leaves the team.

### PR DESCRIPTION
**Pull request by TristanDebrunner**
#### Problem

We need a way to log in to the new datacenter nodes.

#### Summary of Changes

Separate out the list of authorized pubkeys and create `add-datacenter-solana-user-authorized_keys.sh`

Fixes #5440

### CODE REVIEW

File: net/scripts/solana-user-authorized_keys.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+SOLANA_PUBKEYS+=('ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBOk4jgcX/VWSk3j//wXeIynSQjsOt+AjYXM/XZUMa7R1Q8lfIJGK/qHLBP86CMXdpyEKJ5i37QLYOL+0VuRy0CI=')
+
+SOLANA_USERS+=('jack')
+SOLANA_PUBKEYS+=('ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEB6YLY4oCfm0e1qPswbzryw0hQEMiVDcUxOwT4bdBbui/ysKGQlVY8bO6vET1Te8EYHz5W4RuPfETbcHmw6dr4=')
+
+SOLANA_USERS+=('trent')
+SOLANA_PUBKEYS+=('ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEZC/APgZTM1Y/EfNnCHr+BQN+SN4KWfpyGkwMg+nXdC trent@fry')
+
+SOLANA_USERS+=('tristan')
+SOLANA_PUBKEYS+=('ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0+3//c4pSjMzysddEQFiRWwow8iyPuwkAJ2CDSuAFHzZ22XBdzghwNyJz97suaiefOc5hAgPWTeM8p0NpHsPItgRlwqljxDrQDKEW2K79j3/rnS8h0OJWAnoJ5bX8ft1h+C3mtUJtOWN67wGSpIOIxo1j/zHJHKJZSMmSxEtd4FzdUsMdezEeF1rDak2z4UlpEX1Y7rir8I2D4uvpLRBBCyW49NX/dAv+cpNoJD6eb1Kt85bfB3VTO5Qv8gsMnAruM+OQ2GnYUE1OgHR3miPSyQ7S0NvwKTe/k8L24dlNzy6Hw+jlQiqE/2O5llBZc07I878WyqMMV+Wxk9nQFyWT tristan@TristanSolanaMacBook.local')
```
Comment (mvines) [inline net/scripts/solana-user-authorized_keys.sh]: RSA keys 😱

Comment (TristanDebrunner) [reply net/scripts/solana-user-authorized_keys.sh]: If I'm reading [this](https://tools.ietf.org/id/draft-ietf-curdle-ssh-kex-sha2-10.html#rfc.section.3.28) correctly, the 2048bit rsa keys should be acceptable, but I do have to change mine.

Comment (t-nelson) [reply net/scripts/solana-user-authorized_keys.sh]: IIRC 2048-bit RSA is more of a *minimum*, 3072 *recommended*, 4096+ *paranoid*.

@mvines is probably complaining about them being a mile long, though :wink:

File: net/scripts/solana-user-authorized_keys.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+SOLANA_USERS+=('sathish')
+SOLANA_PUBKEYS+=('ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBGqZAwAZeBl0buOMz4FpUYrtpwk1L5aGKlbd7lI8dpbSx5WVRPWCVKhWzsGMtDUIfmozdzJouk1LPyihghTDgsE=')
+
+SOLANA_USERS+=('carl')
+SOLANA_PUBKEYS+=('ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBOk4jgcX/VWSk3j//wXeIynSQjsOt+AjYXM/XZUMa7R1Q8lfIJGK/qHLBP86CMXdpyEKJ5i37QLYOL+0VuRy0CI=')
+
+SOLANA_USERS+=('jack')
+SOLANA_PUBKEYS+=('ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBEB6YLY4oCfm0e1qPswbzryw0hQEMiVDcUxOwT4bdBbui/ysKGQlVY8bO6vET1Te8EYHz5W4RuPfETbcHmw6dr4=')
+
+SOLANA_USERS+=('trent')
```
Comment (t-nelson) [inline net/scripts/solana-user-authorized_keys.sh]: So if I want to add another key for my workstation, I duplicate the `SOLANA_USERS` entry as well?

File: net/scripts/add-datacenter-solana-user-authorized_keys.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+cd "$(dirname "$0")"
+
+# shellcheck source=net/scripts/solana-user-authorized_keys.sh
+source solana-user-authorized_keys.sh
+
+# solana-user-authorized_keys.sh defines the public keys for users that should
+# automatically be granted access to ALL datacenter nodes.
+for i in "${!SOLANA_USERS[@]}"; do
+  echo "environment=\"SOLANA_USER=${SOLANA_USERS[i]}\" ${SOLANA_PUBKEYS[i]}" >> /solana-authorized_keys
```
Comment (t-nelson) [inline net/scripts/add-datacenter-solana-user-authorized_keys.sh]: Should the temp file be in `/` ?

File: net/scripts/add-testnet-solana-user-authorized_keys.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+cd "$(dirname "$0")"
+
+# shellcheck source=net/scripts/solana-user-authorized_keys.sh
+source solana-user-authorized_keys.sh
+
+# solana-user-authorized_keys.sh defines the public keys for users that should
+# automatically be granted access to ALL testnets.
+for key in "${SOLANA_PUBKEYS[@]}"; do
+  echo "$key" >> sample_out.txt #/solana-authorized_keys
```
Comment (t-nelson) [inline net/scripts/add-testnet-solana-user-authorized_keys.sh]: We want the commented file name here

File: net/scripts/add-datacenter-solana-user-authorized_keys.sh (hunk 3 above)
Comment (TristanDebrunner) [reply net/scripts/add-datacenter-solana-user-authorized_keys.sh]: That was how the testnet scripts did it, and I didn't change it

File: net/scripts/solana-user-authorized_keys.sh (hunk 2 above)
Comment (TristanDebrunner) [reply net/scripts/solana-user-authorized_keys.sh]: Yeah. I'll clarify that in the comment

File: net/scripts/solana-user-authorized_keys.sh (hunk 1 above)
Comment (TristanDebrunner) [reply net/scripts/solana-user-authorized_keys.sh]: If I'm reading [this](https://tools.ietf.org/id/draft-ietf-curdle-ssh-kex-sha2-10.html#rfc.section.3.28) right, the 2048bit keys should be ok, but I do have to change mine

File: net/scripts/add-datacenter-solana-user-authorized_keys.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+done
+
+sudo -u solana bash -c "
+  cat /solana-authorized_keys >> /home/solana/.ssh/authorized_keys
```
Comment (t-nelson) [inline net/scripts/add-datacenter-solana-user-authorized_keys.sh]: If we make this a `mv`, we don't need to mess with cleaning up the temp file ('cause it's created by an append redirect).

**Pull request merged**